File: ClickServerWeb/src/com/pku/ss/egl/libs/BootControllerLib.java
package com.pku.ss.egl.libs;
import org.eclipse.edt.javart.resources.*;
import org.eclipse.edt.javart.*;
import com.pku.ss.egl.et.BootComputer;
import org.eclipse.edt.runtime.java.eglx.lang.EAny;
import java.lang.Object;
@SuppressWarnings("unused")
@javax.xml.bind.annotation.XmlRootElement(name="BootControllerLib")
public class BootControllerLib extends ExecutableBase {
	private static final long serialVersionUID = 10L;
	private BootComputer bc;
	public BootControllerLib() {
		super();
	}
	{
		ezeInitialize();
	}
	public void ezeInitialize() {
		bc = new BootComputer();
	}
	public BootComputer getBc() {
		return bc;
	}
	public void setBc(BootComputer ezeValue) {
		bc = ezeValue;
	}
	public void sleep() {
		bc.sleep();
	}
	public void shutDown() {
		bc.shutDown();
	}
	public void lock() {
		bc.lock();
	}
	public void logOff() {
		bc.logOff();
	}
	public void hibernate() {
		bc.hibernate();
	}
	public void reboot() {
		bc.reboot();
	}
}

File: ClickServer/generatedJavaScript/com/pku/ss/egl/libs/BootControllerLib.js
define([],function(){
	egl.defineRUILibrary('com.pku.ss.egl.libs', 'BootControllerLib',
	{
		'eze$$fileName': 'com/pku/ss/egl/libs/BootControllerLib.egl',
		'eze$$runtimePropertiesFile': 'com.pku.ss.egl.libs.BootControllerLib',
			"constructor": function() {
				if(egl.com.pku.ss.egl.libs.BootControllerLib['$inst']) return egl.com.pku.ss.egl.libs.BootControllerLib['$inst'];
				egl.com.pku.ss.egl.libs.BootControllerLib['$inst']=this;
			}
			,
			"eze$$setEmpty": function() {
				this.bc = new ();
			}
			,
			"eze$$setInitial": function() {
				if(egl.com.pku.ss.egl.libs.BootControllerLib['$inst']) return egl.com.pku.ss.egl.libs.BootControllerLib['$inst'];
				egl.com.pku.ss.egl.libs.BootControllerLib['$inst']=this;
				this.eze$$setEmpty();
			}
			,
			"eze$$getAnnotations": function() {
				if(this.annotations === undefined){
					this.annotations = {};
					this.annotations["XMLRootElement"] = new egl.eglx.xml.binding.annotation.XMLRootElement("BootControllerLib", null, false);
				}
				return this.annotations;
			}
			,
			"eze$$getFieldInfos": function() {
				if(this.fieldInfos === undefined){
					var annotations;
					this.fieldInfos = new Array();
				}
				return this.fieldInfos;
			}
			,
			"sleep": function() {
				this.bc.sleep();
			}
			,
			"shutDown": function() {
				this.bc.shutDown();
			}
			,
			"lock": function() {
				this.bc.lock();
			}
			,
			"logOff": function() {
				this.bc.logOff();
			}
			,
			"hibernate": function() {
				this.bc.hibernate();
			}
			,
			"reboot": function() {
				this.bc.reboot();
			}
			,
			"getBc": function() {
				return bc;
			}
			,
			"setBc": function(ezeValue) {
				this.bc = ezeValue;
			}
			,
			"toString": function() {
				return "[BootControllerLib]";
			}
		}
	);
});
